### DISCUSSION TITLE
ROS is not initialized. The status indicator LED is remaining white.

### DISCUSSION THREAD

**Discussion by haguremetaru** [HumanSupportRobot/QandA#105] [Q&A] [answered]
ROS on the HSR does not initialize at all on starting/restarting the HSR. After turning on the HSR, the status indicator LED remains flashing with a white light.

When I run “rostopic list”, the output is “ERROR: Unable to communicate with master!"

I have seen similar discussions(https://github.com/orgs/HumanSupportRobot/discussions/95) in the past and have tried various things but have not been able to solve the problem.

systemctl status docker.hsrb.roscore.service
systemctl status docker.hsrb.robot.service
docker logs -f docker.hsrb.roscore.service
docker logs -f docker.hsrb.robot.service

The results of executing the above four commands are shown below.

systemctl status docker.hsrb.roscore.service
> administrator@hsrb26:~$ systemctl status docker.hsrb.roscore.service
● docker.hsrb.roscore.service - HSRB Core service
     Loaded: loaded (/etc/systemd/system/docker.hsrb.roscore.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2024-06-07 06:49:49 UTC; 9s ago
    Process: 54626 ExecStartPre=/usr/bin/install -d -m755 -o ros -g ros /var/run/ros (code=exited, status=0/SUCCESS)
    Process: 54627 ExecStartPre=/usr/bin/install -d -m755 -o ros -g ros /var/run/ros/workspace (code=exited, status=0/SUCCESS)
    Process: 54628 ExecStartPre=/bin/bash -c . /etc/opt/tmc/robot/roshost.sh; /bin/systemctl set-environment $(get_roshost) (code=exited, status=0/SUCCESS)
    Process: 54637 ExecStartPre=/usr/bin/docker rm -f docker.hsrb.roscore.service (code=exited, status=0/SUCCESS)
    Process: 54674 ExecStartPre=/bin/sh -c exec /usr/bin/docker run     --name docker.hsrb.roscore.service      --detach        --tty   --volume=${WORKSPACE}:${WORKSPACE}:>
   Main PID: 54757 (docker)
      Tasks: 9 (limit: 19036)
     Memory: 11.1M
     CGroup: /system.slice/docker.hsrb.roscore.service
             └─54757 /usr/bin/docker logs -f docker.hsrb.roscore.service
 6月 07 06:49:50 hsrb26.local docker.hsrb.roscore.service[54757]: roslaunch.core.RLException: unable to create directory for log file [/var/log/ros/20240607-064950_22f9be4>
 6月 07 06:49:50 hsrb26.local docker.hsrb.roscore.service[54757]: Please check permissions.
 6月 07 06:49:50 hsrb26.local docker.hsrb.roscore.service[54757]:
 6月 07 06:49:50 hsrb26.local docker.hsrb.roscore.service[54757]: [master] ERROR: unable to configure logging [unable to create directory for log file [/var/log/ros/202406>
 6月 07 06:49:50 hsrb26.local docker.hsrb.roscore.service[54757]: Please check permissions.]
 6月 07 06:49:50 hsrb26.local docker.hsrb.roscore.service[54757]: [master] ERROR: unable to configure logging [unable to create directory for log file [/var/log/ros/202406>
 6月 07 06:49:50 hsrb26.local docker.hsrb.roscore.service[54757]: Please check permissions.]
 6月 07 06:49:50 hsrb26.local docker.hsrb.roscore.service[54757]: process[master]: started with pid [52]
 6月 07 06:49:50 hsrb26.local docker.hsrb.roscore.service[54757]: [master] process has died [pid 52, exit code 1, cmd python3 /opt/ros/noetic/lib/hsrb_bringup/tmc_rosmaste>
 6月 07 06:49:50 hsrb26.local docker.hsrb.roscore.service[54757]: [master] process has died [pid 52, exit code 1, cmd python3 /opt/ros/noetic/lib/hsrb_bringup/tmc_rosmaste>
lines 1-24/24 (END)

systemctl status docker.hsrb.robot.service
> administrator@hsrb26:~$ systemctl status docker.hsrb.robot.service
● docker.hsrb.robot.service - HSRB Robot service
     Loaded: loaded (/etc/systemd/system/docker.hsrb.robot.service; enabled; vendor preset: enabled)
     Active: activating (auto-restart) since Fri 2024-06-07 06:50:50 UTC; 6s ago
    Process: 57896 ExecStartPre=/bin/rm -f ${ROS_HOME}/map (code=exited, status=0/SUCCESS)
    Process: 57897 ExecStartPre=/bin/bash -c . /etc/opt/tmc/robot/roshost.sh; /bin/systemctl set-environment $(get_roshost) (code=exited, status=0/SUCCESS)
    Process: 57906 ExecStartPre=/usr/bin/docker rm -f docker.hsrb.robot.service (code=exited, status=0/SUCCESS)
    Process: 57943 ExecStartPre=/bin/sh -c exec /usr/bin/docker run     --name docker.hsrb.robot.service        --detach        --privileged --volume=/dev:/dev         --t>
    Process: 58030 ExecStart=/usr/bin/docker logs -f docker.hsrb.robot.service (code=exited, status=0/SUCCESS)
    Process: 58118 ExecStop=/usr/bin/docker stop docker.hsrb.robot.service (code=exited, status=1/FAILURE)
    Process: 58151 ExecStopPost=/usr/bin/docker stop docker.hsrb.robot.service (code=exited, status=1/FAILURE)
   Main PID: 58030 (code=exited, status=0/SUCCESS)

docker logs -f docker.hsrb.roscore.service
> administrator@hsrb26:~$ docker logs -f docker.hsrb.roscore.service
WARNING: cannot create log directory [/var/log/ros/20240607-065141_655a1d50-249a-11ef-a75e-003064445192]. Please set ROS_LOG_DIR to a writable location.
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is <1GB.
started roslaunch server http://10.40.20.141:43101/
SUMMARY
========
PARAMETERS

>  * /hsrb/diagnostic_aggregator/analyzers/cpu/analyzers/computer/find_and_remove_prefix: tmc_computer_moni...
>  * /hsrb/diagnostic_aggregator/analyzers/cpu/analyzers/computer/path: Computer
>  * /hsrb/diagnostic_aggregator/analyzers/cpu/analyzers/computer/timeout: 105.0
>  * /hsrb/diagnostic_aggregator/analyzers/cpu/analyzers/computer/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/cpu/analyzers/file/find_and_remove_prefix: tmc_computer_moni...
>  * /hsrb/diagnostic_aggregator/analyzers/cpu/analyzers/file/path: File
>  * /hsrb/diagnostic_aggregator/analyzers/cpu/analyzers/file/timeout: 0
>  * /hsrb/diagnostic_aggregator/analyzers/cpu/analyzers/file/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/cpu/path: Computer
>  * /hsrb/diagnostic_aggregator/analyzers/cpu/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/joints/num_items: 11
>  * /hsrb/diagnostic_aggregator/analyzers/joints/path: Joints
>  * /hsrb/diagnostic_aggregator/analyzers/joints/startswith: Joint
>  * /hsrb/diagnostic_aggregator/analyzers/joints/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/power/analyzers/battery/find_and_remove_prefix: hsrb/sanyo_battery:
>  * /hsrb/diagnostic_aggregator/analyzers/power/analyzers/battery/path: Battery
>  * /hsrb/diagnostic_aggregator/analyzers/power/analyzers/battery/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/power/path: Power
>  * /hsrb/diagnostic_aggregator/analyzers/power/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/bumper/path: Bumper
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/bumper/startswith: hsrb_bumper
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/bumper/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/analyzers/hand_camera/find_and_remove_prefix: hsrb/hand_camera/...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/analyzers/hand_camera/path: HandCamera
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/analyzers/hand_camera/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/analyzers/head_center_camera/find_and_remove_prefix: hsrb/head_center_...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/analyzers/head_center_camera/path: HeadCenterCamera
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/analyzers/head_center_camera/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/analyzers/head_rgbd_sensor/find_and_remove_prefix: hsrb/head_rgbd_se...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/analyzers/head_rgbd_sensor/path: HeadRGBDSensor
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/analyzers/head_rgbd_sensor/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/analyzers/head_stereo_camera/find_and_remove_prefix: hsrb/stereo_camer...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/analyzers/head_stereo_camera/path: HeadStereoCamera
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/analyzers/head_stereo_camera/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/path: Camera
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/camera/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/ft_sensor/find_and_remove_prefix: hsrb/ft_sensor_di...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/ft_sensor/path: FtSensor
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/ft_sensor/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/io_monitor/find_and_remove_prefix: hsrb/hsrb_io_moni...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/io_monitor/path: IO
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/io_monitor/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/urg/find_and_remove_prefix: hsrb/urg_node:
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/urg/path: Hokuyo
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/analyzers/urg/type: diagnostic_aggreg...
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/path: Sensors
>  * /hsrb/diagnostic_aggregator/analyzers/sensors/type: diagnostic_aggreg...
>  * /hsrb/diagnostics_journald/forward: True
>  * /hsrb/diagnostics_journald/syslog_identifier: rosdiag
>  * /hsrb/diagnostics_journald/throttle_period: 30.0
>  * /hsrb/rosout_journald/forward: False
>  * /hsrb/rosout_journald/syslog_identifier: rosout
>  * /rosdistro: noetic
>  * /rosversion: 1.15.14
> NODES
>   /hsrb/
>     diagnostic_aggregator (diagnostic_aggregator/aggregator_node)
>     diagnostics_journald (tmc_journald/diagnostics)
>     rosout_journald (tmc_journald/rosout)
>     runtime_logger (rosbag/record)
> auto-starting new master
> Traceback (most recent call last):
>   File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/nodeprocess.py", line 250, in _configure_logging
>     os.makedirs(log_dir)
>   File "/usr/lib/python3.8/os.py", line 223, in makedirs
>     mkdir(name, mode)
> PermissionError: [Errno 13] Permission denied: '/var/log/ros/20240607-065141_655a1d50-249a-11ef-a75e-003064445192'
> During handling of the above exception, another exception occurred:
> Traceback (most recent call last):
>   File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/nodeprocess.py", line 305, in start
>     logfileout, logfileerr = self._configure_logging()
>   File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/nodeprocess.py", line 253, in _configure_logging
>     raise RLException("unable to create directory for log file [%s].\nPlease check permissions."%log_dir)
> roslaunch.core.RLException: unable to create directory for log file [/var/log/ros/20240607-065141_655a1d50-249a-11ef-a75e-003064445192].
> Please check permissions.
> [master] ERROR: unable to configure logging [unable to create directory for log file [/var/log/ros/20240607-065141_655a1d50-249a-11ef-a75e-003064445192].
> Please check permissions.]
> [master] ERROR: unable to configure logging [unable to create directory for log file [/var/log/ros/20240607-065141_655a1d50-249a-11ef-a75e-003064445192].
> Please check permissions.]
> process[master]: started with pid [53]
> [master] process has died [pid 53, exit code 1, cmd python3 /opt/ros/noetic/lib/hsrb_bringup/tmc_rosmaster --core -p 11311].
> [master] process has died [pid 53, exit code 1, cmd python3 /opt/ros/noetic/lib/hsrb_bringup/tmc_rosmaster --core -p 11311].
> RLException: ERROR: could not contact master [http://localhost:11311/]
> RLException: ERROR: could not contact master [http://localhost:11311/]
> The traceback for the exception was written to the log file
> The traceback for the exception was written to the log file
> Traceback (most recent call last):
>   File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/__init__.py", line 347, in main
>     p.start()
>   File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/parent.py", line 316, in start
>     self.runner.launch()
>   File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/launch.py", line 676, in launch
>     self._setup()
>   File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/launch.py", line 651, in _setup
>     launched = self._launch_master()
>   File "/opt/ros/noetic/lib/python3/dist-packages/roslaunch/launch.py", line 429, in _launch_master
>     raise RLException("ERROR: could not contact master [%s]"%m.uri)
> roslaunch.core.RLException: ERROR: could not contact master [http://localhost:11311/]

docker logs -f docker.hsrb.robot.service
> administrator@hsrb26:~$ docker logs -f docker.hsrb.robot.service
Error response from daemon: No such container: docker.hsrb.robot.service

I checked /etc/systemd/system/docker.hsrb.roscore.service and it is fine.
I have also confirmed that Docker Image and Environment Files are present.

**Comment by yutaro-ishida** [comment, marked as answer]
ROS master can't be launched because it lacks permission to create a ROS log directory under `／var／log`. Is there anything that comes to mind about the above?

**Reply by haguremetaru** [reply]
Thank you. The issue has been resolved. I accidentally deleted the folder.
